### ISSUE TITLE
Pros and cons of various Machine Learning algorithms

### ISSUE THREAD

**Issue by WatershedXiaolan** [WatershedXiaolan/Xiaolan-s-Gitbook#210]
### SVM 
**Pros**
- Works well in high dimension 
- Outliers have less impact 
**Cons**
- Slow
- Poor performance with Overlapped classes 
- Hyper param tuning: regularization and kernel functions 
**Applications**
Classification with many features 

### Naive Bayes 
**Pros**
- Real time prediction 
- scalable with large dataset 
- Insensitive to irrelevant features.
- multi-class prediction 
**Cons**
- Independence of the features
- Training data should represent the population well

### Logistic Regression
**Pros**
- Simple and fast 
- Feature scaling is not needed
- Tuning of hyperparameters not needed.
**Cons**
- Poor performance on non-linear data
- Need regularization 
- Too simple

### Random Forest
**Pros**
- Good Performance on Imbalanced datasets
- Handling of huge amount of data
- Little impact of outliers
- Good handling of missing data
- No problem of overfitting
- Useful to extract feature importance
**Cons**
- Prediction accuracy on complex problems is usually inferior to gradient-boosted trees

### XGBoost
**Pros**
- Less feature engineering required
- Feature Importance 
- Fast 
- Not easy to overfit
**Cons**
- Tune parameters 

### KNN
**Pros**
- No assumption about data
- Multi-class
- One Hyper Parameter
**Cons**
- slow 
- scaling is a must
- Does not work well on Imbalanced data
- Sensitive to outliers.

### Logistic Regression
**Pros**
**Cons**
